### PR TITLE
[DOC] Add comparison of meaning differences of GLM between different libraries

### LINKED ISSUE #2599: Add explanations for 1st and 2nd level for glm
Add a few lines somewhere in the doc to explain what are `glm` `first` and `second levels` which might not be obvious for researchers from other fields. 
Note: We could also put a link to these explanations in the html reports.

### PR DESCRIPTION
**Pull request by tpremrud**
- Closes #2599 

Changes proposed in this pull request:

- Added `meaning_difference.rst` as a documentation file for comparing GLM analysis levels between different libraries (i.e., FSL and SPM)
- Modified the index page of GLM documentation module to include the documentation structure (i.e., drop-down list) for the `meaning_difference.rst` page.

**Note**: I am still uncertain if some of the wording and content are redundant (e.g., runs vs sessions, which I am certain are the same, but I included for clarity); please feel free to make any suggestions! :)

### CODE REVIEW

File: doc/glm/meaning_difference.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		* First-level: analyze one session for one subject
+		* Second-level: analyze across sessions for one subject
+		* Third-level: group-level analysis
+	
+These differences can be put in the table below:
+
++----------------------------------------------+----------------+------------------+---------------+
+| Analyze                                      | nilearn        | SPM              | FSL           |
++==============================================+================+==================+===============+
+| One session, one subject                     | N\/A           | N\/A             | First\-level  |
```
Comment (tsalo) [inline doc/glm/meaning_difference.rst:22]:
```suggestion
| One session, one subject                     | First-level           | First-level             | First\-level  |
```

File: doc/glm/meaning_difference.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
+.. _meaning_difference:
+
+==================================================
+Difference in meanings between different toolboxes
+==================================================
+
+.. topic:: **Page summary**
+
+	* SPM has the same meaning of analysis levels, compared to nilearn's models
+		* First-level: analyze across sessions
```
Comment (tsalo) [inline doc/glm/meaning_difference.rst:10]:
```suggestion
		* First-level: analyze across runs
```
Following BIDS convention, runs and sessions are different things, and you typically don't analyze across sessions in a first-level model.

Comment (tpremrud) [reply doc/glm/meaning_difference.rst:10]: Agreed, great catch, thank you!!!

**Comment by Remi-Gau** [comment]
Thanks for starting this @tpremrud: I can already tell this will be usefull!

Couple of general comments.

I would suggest mentioning that we use the BIDS definitions of "run" in this document (and in general we are trying to move to make this use consistent in nilearn).

Also as far as I can tell all the mentions of the word "session" in your text should be replaced by "run".

https://bids-specification.readthedocs.io/en/latest/common-principles.html#definitions

> Run - An uninterrupted repetition of data acquisition that has the same acquisition parameters and task (however events can change from run to run due to different subject response or randomized nature of the stimuli). Run is a synonym of a data acquisition. Note that "uninterrupted" may look different by modality due to the nature of the recording. For example, in [MRI](https://bids-specification.readthedocs.io/en/latest/modality-specific-files/magnetic-resonance-imaging-data.html) or [MEG](https://bids-specification.readthedocs.io/en/latest/modality-specific-files/magnetoencephalography.html), if a subject leaves the scanner, the acquisition must be restarted. For some types of [PET](https://bids-specification.readthedocs.io/en/latest/modality-specific-files/positron-emission-tomography.html) acquisitions, a subject may leave and re-enter the scanner without interrupting the scan.

I do not think we need the whole definition but at least the main bit and then a link to the BIDS glossary:

- https://bids-specification.readthedocs.io/en/latest/glossary.html#run-common_principles

File: doc/glm/meaning_difference.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+SPM uses the same notation as nilearn for analysis levels, with a note that a session still refers to an imaging session or a run, and within a run there could be multiple conditions (e.g., congruent and incongruent).
+In this case, `SPM`_ provided `tutorials`_ and documentation, including `lectures`_, which one could learn to analyze their own fMRI data with the meaning of analysis levels being as follows:
+
+	* `First-level analysis in SPM`_: Analyze across sessions for a subject (i.e., more than one session of one subject)
+	* `Second-level analysis in SPM`_: Analyze across several subjects (i.e., more than one subject with one or more sessions per subject). This is also known as **group-level analysis** which test if the average estimate across subjects is statistically significant.
+
+.. _SPM: https://www.fil.ion.ucl.ac.uk/spm/docs/
+.. _tutorials: https://www.fil.ion.ucl.ac.uk/spm/docs/tutorials/
+.. _lectures: https://www.fil.ion.ucl.ac.uk/spm/docs/courses/fmri_vbm/recordings/glm/
```
Comment (Remi-Gau) [inline doc/glm/meaning_difference.rst:43]: Wonder if we should already start pointing to the new SPM documentation website:
see this VERY recent announcement https://github.com/spm/spm/discussions/25

File: doc/glm/meaning_difference.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+*Table showing the differences of the meaning for level of models between toolboxes\/libraries*
+
+Statistical Parametric Mapping (SPM)
+====================================
+
+SPM uses the same notation as nilearn for analysis levels, with a note that a session still refers to an imaging session or a run, and within a run there could be multiple conditions (e.g., congruent and incongruent).
+In this case, `SPM`_ provided `tutorials`_ and documentation, including `lectures`_, which one could learn to analyze their own fMRI data with the meaning of analysis levels being as follows:
+
+	* `First-level analysis in SPM`_: Analyze across sessions for a subject (i.e., more than one session of one subject)
```
Comment (Remi-Gau) [inline doc/glm/meaning_difference.rst:38]: I suspect that we should mention somewhere that it is typical in the SPM workflow to put all runs in a single design matrix, where as nilearn typically will give you one design matrix per run.

Comment (tpremrud) [reply doc/glm/meaning_difference.rst:38]: Gotcha @Remi-Gau , that's a really great point! However, I'm wondering if I understand the concept correctly, so if there are different conditions, do they still count as a single run if the imaging acquisition is continuous, uninterrupted, and comprise of the same parameters?

I'm looking at [nilearn example](https://nilearn.github.io/dev/auto_examples/04_glm_first_level/plot_design_matrix.html) it seems like there are different conditions within a design matrix, but you said that nilearn gives one design matrix per run, so I'm just not sure if I understand the concept of run correctly as I should or not.

But the [SPM](https://andysbrainbook.readthedocs.io/en/latest/SPM/SPM_Short_Course/SPM_Statistics/SPM_06_Stats_Running_1stLevel_Analysis.html) really does have a design matrix that goes through all the runs in one matrix.

Thank you!!

File: doc/glm/meaning_difference.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
+.. _meaning_difference:
+
+==================================================
+Difference in meanings between different toolboxes
+==================================================
+
+.. topic:: **Page summary**
+
+	* SPM has the same meaning of analysis levels, compared to nilearn's models
```
Comment (bthirion) [inline doc/glm/meaning_difference.rst:9]:
```suggestion
	* SPM has the same meaning of analysis levels, compared to Nilearn's models
```

File: doc/glm/meaning_difference.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+| More than or equal to one session,           | Second\-level  | Second\-level    | Third\-level  |
+| more than one subject                        |                |                  |               |
++----------------------------------------------+----------------+------------------+---------------+
+
+*Table showing the differences of the meaning for level of models between toolboxes\/libraries*
+
+Statistical Parametric Mapping (SPM)
+====================================
+
+SPM uses the same notation as nilearn for analysis levels, with a note that a session still refers to an imaging session or a run, and within a run there could be multiple conditions (e.g., congruent and incongruent).
```
Comment (bthirion) [inline doc/glm/meaning_difference.rst:35]:
```suggestion
SPM uses the same notation as Nilearn for analysis levels, with a note that a session still refers to an imaging session or a run, and within a run there could be multiple conditions (e.g., congruent and incongruent).
```

**Review by bthirion — COMMENTED** [review]
LGTM overall.

**Comment by Remi-Gau** [comment]
will clean and fix CI on this one

**Review by Remi-Gau — APPROVED** [review]

File: CITATION.cff [hunk 7]
```diff
@@ -710,6 +710,9 @@ authors:
   - given-names: Tharun
     family-names: K
     website: https://github.com/tharun634
+  - given-names: Thiti
+    family-names: Premrudeepreechacharn
+    website: https://tpremrud.github.io/
```
Comment (Remi-Gau) [inline CITATION.cff:715]: @tpremrud 
I have added these details about you as a contributor.
Let me know if they should be updated.

File: doc/glm/glm_intro.rst [hunk 8]
```diff
@@ -20,6 +20,22 @@ The signal picked up by the MRI scanner is sensitive to these modifications of t
 To record cerebral activity during functional runs,
 the scanner is tuned to detect this "Blood Oxygen Level Dependent" (:term:`BOLD`) signal.
 
+.. admonition:: What do we mean by a ``run``?
+    :class: important
+
+    In Nilearn we try to stick to the :term:`BIDS` definition of run.
```
Comment (Remi-Gau) [inline doc/glm/glm_intro.rst:26]: @bthirion 
felt like it was good the define what we mean by run.

File: doc/glm/meaning_difference.rst [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    Nilearn will instead create one design matrix per run,
+    and run one model per run
+    (see for example :ref:`this report with 2 runs <two_runs_glm>`).
+    In Nilearn, to compute summary statistics across runs
+    you can use the method :meth:`~nilearn.glm.first_level.FirstLevelModel.compute_contrast`
+    or the function :func:`~nilearn.glm.compute_fixed_effects`.
+
+    .. seealso::
+
+        :ref:`sphx_glr_auto_examples_04_glm_first_level_plot_two_runs_model.py`
```
Comment (Remi-Gau) [inline doc/glm/meaning_difference.rst:65]: @bthirion 
added this admonition regarding fixed effect analysis: does it make sense to you?

Comment (bthirion) [reply doc/glm/meaning_difference.rst:65]: That's very clear. Thx

**Comment by Remi-Gau** [comment]
will wait so that we can check the HTML preview

**Comment by Remi-Gau** [comment]
for once the linkchecks did its job!!! 

some of the new links were actually broken

**Comment by Remi-Gau** [comment]
OK HTML looks good

merging

thanks @tpremrud and sorry it took so long to get back to this!!!

**Pull request merged**